wordpress-mobile/WordPress-Android
# [Jetpack Social] Enable/disable post settings connections considering shares remaining

Part of #18571 

Should only be merged after https://github.com/wordpress-mobile/WordPress-Android/pull/18861

To test:
1 - Log in with a .com account;
2 - Enable JetpackSocialFeatureConfig feature flag;
3 - Make sure you have at least one social sharing account connected;
4 - Select a SH site with Jetpack plugin or Jetpack Social plugin installed that has more accounts than remaining shares (e.g. 2 social sharing accounts connected and 1 remaining share);
5 - Open post list;
6 - Select a draft or create a new one;
7 - Tap the overflow menu and select "Post settings";
8 - Scroll down to the "Social Sharing" section: all the connections should be toggled off in this situation;
9 - Tap on connections to enable/disable them: when the number of remaining shares reaches 0 the remaining, the accounts that are not toggled on should be disabled (which means users can't interact with them). If you toggle off a connection, the other connections should be enabled again (and it's possible to interact with them).

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Waiting for https://github.com/wordpress-mobile/WordPress-Android/pull/18866 to be merged so I can update the tests considering this PR.

PR submission checklist:


UI Changes testing checklist:
